ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/helpers/StructHelper.java

```diff
@@ -0,0 +1,223 @@
+
+    public static String getFreeFunctionName(final String name) {
+        return getResponseTypeName(name) + "_free";
+    }
+
+    public static String getFreeFunction(final StructMember structMember) throws ParseException {
+        switch (structMember.getType()) {
+            case "ds3_str*":
+                return "ds3_str_free";
+            // The following primitive types don't require a free
+            case "uint64_t":
+            case "double":
+            case "long":
+            case "int":
+            case "ds3_bool":
+                return "";
+
+            // build the name of the free function for the embedded type
+            case "java.util.Set":
+            case "array":
+            default:
+                return getFreeFunctionName(structMember.getType());
+        }
+    }
+
+    public static String convertDs3ElementType(final Ds3Element element) throws ParseException {
+        switch (element.getType()) {
+            case "boolean":
+                return "ds3_bool";
+            case "java.lang.String":
+            case "java.util.Date":
+            case "java.util.UUID":
+                return "ds3_str*";
+            case "double":
+                return "double";   // ex: 0.82
+            case "java.lang.Long":
+            case "long":
+                return "uint64_t"; // size_t
+            case "java.lang.Integer":
+            case "int":
+                return "int";
+            case "java.util.Set":
+            case "array":
+                return getResponseTypeName(element.getComponentType()) + "**";
+
+            default:
+                return getResponseTypeName(element.getType()) + "*";
+        }
+    }
+
+    public static boolean isPrimitiveDs3Type(final String type){
+        switch (type) {
+            case "ds3_bool":
+            case "ds3_str*":
+            case "double":
+            case "uint64_t":
+            case "int":
+                return true;
+            default:
+                // any complex sub element such as "com.spectralogic.s3.server.domain.UserApiBean"
+                return false;
+        }
+    }
+
+    public static boolean isPrimitive(final Struct struct) {
+        for (final StructMember member : struct.getVariables()) {
+            if (!isPrimitiveDs3Type(member.getType())) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static ImmutableList<StructMember> convertDs3Elements(final ImmutableList<Ds3Element> elementsList) throws ParseException {
+        final ImmutableList.Builder<StructMember> builder = ImmutableList.builder();
+        for (final Ds3Element currentElement : elementsList) {
+            builder.add(new StructMember(convertDs3ElementType(currentElement), getNameUnderscores(currentElement.getName())));
+        }
+        return builder.build();
+    }
+
+    public static boolean containsExistingStructs(final Struct struct, final Set<String> existingStructs) {
```

Comment:
It doesn't have to be, and he's using this for a Set anyway.  Though I would argue that since this is only used by one class, that it should be moved to that same class, and made private